### PR TITLE
allowing CompactFiles to return new file names

### PR DESCRIPTION
**Pull request by nivdayan**
This is a small API extension to allow the CompactFiles method to return the names of files that were created during the compaction.

### CODE REVIEW

File: db/db_impl_compaction_flush.cc [hunk 1]
```diff
@@ -680,6 +682,12 @@ Status DBImpl::CompactFilesImpl(
     }
   }
 
+  if (output_file_names != NULL) {
```
Comment (siying) [inline db/db_impl_compaction_flush.cc]: Please use `nullptr`.

File: db/db_impl_compaction_flush.cc [hunk 2]
```diff
@@ -680,6 +682,12 @@ Status DBImpl::CompactFilesImpl(
...
+    for (const auto newf : c->edit()->GetNewFiles()) {
+      (*output_file_names).push_back(TableFileName(immutable_db_options_.db_paths, newf.second.fd.GetNumber(), newf.second.fd.GetPathId()) );
```
Comment (siying) [inline db/db_impl_compaction_flush.cc]: Our convention is 80-char limit.

File: db/db_test.cc [hunk 3]
```diff
@@ -2411,7 +2411,8 @@ class ModelDB : public DB {
       const CompactionOptions& /*compact_options*/,
       ColumnFamilyHandle* /*column_family*/,
       const std::vector<std::string>& /*input_file_names*/,
-      const int /*output_level*/, const int /*output_path_id*/ = -1) override {
+      const int /*output_level*/, const int /*output_path_id*/ = -1,
+      std::vector<std::string>* const /*output_file_names*/ = NULL) override {
```
Comment (siying) [inline db/db_test.cc]: `nullptr`

File: include/rocksdb/db.h [hunk 4]
```diff
@@ -814,14 +814,17 @@ class DB {
       const CompactionOptions& compact_options,
       ColumnFamilyHandle* column_family,
       const std::vector<std::string>& input_file_names,
-      const int output_level, const int output_path_id = -1) = 0;
+      const int output_level, const int output_path_id = -1,
+      std::vector<std::string>* const output_file_names = NULL) = 0;
```
Comment (siying) [inline include/rocksdb/db.h]: `nullptr`.

**Comment by nivdayan** [comment]
Thanks very much for the stylistic comments, which are now fixed. Please let me know if there is anything else, and thanks again!

**Pull request closed without merging**